idemp-base/src/org/bandahealth/idempiere/base/process/UpdateExpiredStockProcess.java

```diff
@@ -0,0 +1,96 @@
+package org.bandahealth.idempiere.base.process;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+
+import org.compiere.model.MAttributeSetInstance;
+import org.compiere.model.MDocType;
+import org.compiere.model.MStorageOnHand;
+import org.compiere.model.Query;
+import org.compiere.process.ProcessInfoParameter;
+import org.compiere.process.StorageCleanup;
+import org.compiere.process.SvrProcess;
+
+/**
+ * This process resets expired stocks to zero for a given client
+ * 
+ * @author clinton
+ *
+ */
+
+public class UpdateExpiredStockProcess extends SvrProcess {
+
+    private final String PROCESS_NAME = this.getClass().getName();
+    public static final String PARAMETERNAME_C_DocType_ID = "C_DocType_ID";
+    
+    @Override
+    protected void prepare() {
+    }
+
+    @Override
+    protected String doIt() throws Exception {
+        log.log(Level.INFO, "START " + PROCESS_NAME);
+
+        int count = 0;
+        //Get all Attribute Set Instance that are expired for this client
+        String attributeWhereClause = MAttributeSetInstance.COLUMNNAME_GuaranteeDate  + " < now()";
+        List<MAttributeSetInstance> expiredAttributes = new Query(getCtx(), MAttributeSetInstance.Table_Name, attributeWhereClause, get_TrxName())
+                .setClient_ID()
+                .setOnlyActiveRecords(true)
+                .list();
+        for (MAttributeSetInstance expiredAttribute : expiredAttributes) {
+            //Get all stocks that have this expired attribute
+            String whereClause = MStorageOnHand.COLUMNNAME_M_AttributeSetInstance_ID + "=?";
+            List<MStorageOnHand> stocks = new Query(getCtx(), MStorageOnHand.Table_Name, whereClause, get_TrxName())
+                    .setClient_ID()
+                    .setParameters(expiredAttribute.getM_AttributeSetInstance_ID())
+                    .setOnlyActiveRecords(true)
+                    .list();
+            for (MStorageOnHand stock : stocks) {
+                UpdateStock.updateStock(stock, BigDecimal.ZERO);
+                count++;
+            }
+        }
+        
+        // Kick off the storage cleanup process
+        StorageCleanup storageCleanupProcess = new StorageCleanup();
+        //Get C_DocType_ID for this client before clearing parameters
+        String whereClause  = "DocBaseType=?";
+        MDocType mDocType = new Query(getCtx(), MDocType.Table_Name, whereClause, null)
+                .setClient_ID()
+                .setParameters(MDocType.DOCBASETYPE_MaterialMovement)
+                .setOnlyActiveRecords(true)
+                .first();
+        if (mDocType != null) {
+            //Clear Parameters
+            clearParameters();
```

Comment:
Nice job on figuring out how to do this! Rather than trying to leverage this current process (the Update Expired Stock Process), can you create a new process to handle this? Pulling from `idemp-rest` `ProcessDBService#runOrderProcess(int)`, something like:
```
		MProcess mprocess = new Query(Env.getCtx(), MProcess.Table_Name, MProcess.COLUMNNAME_AD_Process_UU + "=?",
				get_TrxName()).setOnlyActiveRecords(true).setParameters("8e270648-1d54-46d9-9161-2d0300dd80ff").first();

		MPInstance mpInstance = new MPInstance(mprocess, 0);

		ProcessInfo processInfo = new ProcessInfo(mprocess.getName(), mprocess.getAD_Process_ID());
		processInfo.setAD_PInstance_ID(mpInstance.getAD_PInstance_ID());
		processInfo.setAD_Process_UU(mprocess.getAD_Process_UU());

		processInfo.setParameter(new ProcessInfoParameter[]{
				new ProcessInfoParameter(PARAMETERNAME_C_DocType_ID, mDocType.getC_DocType_ID(), null, null, null)});

		ServerProcessCtl.process(processInfo, null);

		return processInfo.getSummary();
```